src/index.ts

```diff
@@ -1,8 +1,8 @@
 import { Component, createElement, DetailedReactHTMLElement } from 'react'
-import { Props } from './types'
+import { UnionProps } from './types'
 import { isNull, registerImageToLazyLoad } from './lazyload'
 
-abstract class LazyLoad extends Component<Props> {
+abstract class LazyLoad extends Component<UnionProps> {
```

Comment:
I think the types could be better here.
This will be possible after this change? 
```html
<LazyLoadBackgroundImage src="path/image.jpg" alt="my cat" crossOrigin="Anonymous" decoding="async"  />
```
But nether `alt`, `crossOrigin` or `decoding` is valid `div` properties.

I think it's better to do something like:
```ts
abstract class LazyLoad<T extends Props> extends Component<T> {
  ...
}

export class LazyLoadImage extends LazyLoad<ImgProps> {
  ...
}

export class LazyLoadBackgroundImage extends LazyLoad<DivProps> {
  ...
}
```

Or maybe even better to rewrite them to functional components but lets take that in a seperate issue.